omnigibson/utils/object_state_utils.py

```diff
@@ -42,6 +42,48 @@
 })
 
 
+def sample_cuboid_for_predicate(predicate, on_obj, bbox_extent):
```

Comment:
For reviewer context: this refactors the pose sampling part so that we can get a pose without having to go through the placement and stepping.